core/src/main/java/org/mule/component/DefaultComponentLifecycleAdapter.java

```diff
@@ -339,7 +338,7 @@ public Object invoke(MuleEvent event) throws MuleException
         {
             if (componentObject == null)
             {
-                throw new ComponentException(createStaticMessage("componentObject is null"), RequestContext.getEvent(), component);
+                throw new ComponentException(createStaticMessage("componentObject is null"), event, component);
```

Comment:
I've modified the junit to validate this scenario and the event in requestcontext is the correct one. 
I would say that during the catch-exception-strategy handling the event is stored in the requestcontext therefore this is not an issue. I will push the changes to the PR.